benchmark/bench_parse_call.cpp

```diff
@@ -5,13 +5,90 @@ using namespace benchmark;
 using namespace std;
 
 const padded_string EMPTY_ARRAY("[]", 2);
+const char *TWITTER_JSON = SIMDJSON_BENCHMARK_DATA_DIR "twitter.json";
+const char *GSOC_JSON = SIMDJSON_BENCHMARK_DATA_DIR "gsoc-2018.json";
+
+
+
+static void parse_twitter(State& state) {
+  dom::parser parser;
+  padded_string docdata;
+  simdjson::error_code error;
+  padded_string::load(TWITTER_JSON).tie(docdata, error);
+  if(error) {
+      cerr << "could not parse twitter.json" << error << endl;
+      return;
+  }
+  // we do not want mem. alloc. in the loop.
+  error = parser.allocate(docdata.size());
+  if(error) {
+      cout << error << endl;
+      return;
+  }
+  size_t bytes = 0;
+  for (UNUSED auto _ : state) {
+    dom::element doc;
+    bytes += docdata.size();
+    parser.parse(docdata).tie(doc,error);
+    if(error) {
+      cerr << "could not parse twitter.json" << error << endl;
+      return;
+    }
+    benchmark::DoNotOptimize(doc);
+  }
+  state.counters["Bytes"] = benchmark::Counter(
+	        double(bytes), benchmark::Counter::kIsRate,
+	        benchmark::Counter::OneK::kIs1024);
+  state.counters["docs"] = Counter(double(state.iterations()), benchmark::Counter::kIsRate);
+}
+BENCHMARK(parse_twitter)->Repetitions(10)->ComputeStatistics("max", [](const std::vector<double>& v) -> double {
+    return *(std::max_element(std::begin(v), std::end(v)));
+  })->DisplayAggregatesOnly(true);
+
+
+static void parse_gsoc(State& state) {
+  dom::parser parser;
+  padded_string docdata;
+  simdjson::error_code error;
+  padded_string::load(GSOC_JSON).tie(docdata, error);
+  if(error) {
+      cerr << "could not parse gsoc-2018.json" << error << endl;
+      return;
+  }
+  // we do not want mem. alloc. in the loop.
+  error = parser.allocate(docdata.size());
+  if(error) {
+      cout << error << endl;
+      return;
+  }
+  size_t bytes = 0;
+  for (UNUSED auto _ : state) {
+    dom::element doc;
+    bytes += docdata.size();
+    parser.parse(docdata).tie(doc,error);
+    if(error) {
+      cerr << "could not parse gsoc-2018.json" << error << endl;
+      return;
+    }
+    benchmark::DoNotOptimize(doc);
+  }
+  state.counters["Bytes"] = benchmark::Counter(
+	        double(bytes), benchmark::Counter::kIsRate,
+	        benchmark::Counter::OneK::kIs1024);
+  state.counters["docs"] = Counter(double(state.iterations()), benchmark::Counter::kIsRate);
```

Comment:
So that's how this works!

Reply:
> So that's how this works!

Though it is all in the documentation, it took me an embarrassingly long time to make it work correctly. The problem with the Google Benchmark documentation is that they do not have enough practical examples. (Hint! Hint!)